src/components/activity/ActivityTablePanel.svelte

```diff
@@ -1,20 +1,259 @@
+    },
+    tags: {
+      field: 'tags',
+      filter: 'text',
+      headerName: 'Tags',
+      hide: true,
+      resizable: true,
+      sortable: true,
+    }, //string[];
+    type: {
+      field: 'type',
+      filter: 'text',
+      headerName: 'Type',
+      hide: true,
+      resizable: true,
+      sortable: true,
+    },
+  };
+
+  let activityTable: ViewActivityTable;
+  let derivedColumnDefs: ColDef[] = [];
+  let columnMenuItems: ColumnMenuItem[] = [];
+
+  $: activityTable = $view?.definition.plan.activityTables.find(table => table.id === activityTableId);
+  $: derivedColumnDefs = Object.values(defaultColumnDefinitions).map((defaultColumnDef: ColDef) => {
+    const columnDef = activityTable.columnDefs.find((columnDef: ColDef) => columnDef.field === defaultColumnDef.field);
+    if (columnDef) {
+      return {
+        ...defaultColumnDef,
+        ...columnDef,
+        hide: columnDef.hide ?? false,
+      };
+    }
+
+    return defaultColumnDef;
+  });
+  $: columnMenuItems = derivedColumnDefs.map((derivedColumnDef: ColDef) => {
+    const columnState = activityTable.columnStates.find(
+      (columnState: ColumnState) => columnState.colId === derivedColumnDef.field,
+    );
+
+    if (columnState) {
+      return {
+        field: derivedColumnDef.field as keyof Activity,
+        isHidden: columnState?.hide ?? derivedColumnDef.hide ?? false,
+        name: derivedColumnDef.headerName,
+      };
+    }
+
+    return {
+      field: derivedColumnDef.field as keyof Activity,
+      isHidden: true,
+      name: derivedColumnDef.headerName,
+    };
+  });
+
+  function onColumnToggleChange({ detail: { field, isHidden } }: CustomEvent) {
+    const activityColumnStates: ColumnState[] = activityTable?.columnStates ?? [];
+    const existingColumnStateIndex: number = activityColumnStates.findIndex(
+      (columnState: ColumnState) => field === columnState.colId,
+    );
+    if (existingColumnStateIndex >= 0) {
+      viewUpdateActivityTables(
+        {
+          columnStates: [
+            ...activityColumnStates.slice(0, existingColumnStateIndex),
+            {
+              ...activityColumnStates[existingColumnStateIndex],
+              hide: isHidden,
+            },
+            ...activityColumnStates.slice(existingColumnStateIndex + 1),
+          ],
+        },
+        activityTableId,
+      );
+    } else {
+      viewUpdateActivityTables(
+        {
+          columnStates: [
+            ...activityColumnStates,
+            {
+              colId: field,
+              hide: isHidden,
+            },
+          ],
+        },
+        activityTableId,
+      );
+    }
+  }
+
+  function onColumnStateChange({ detail: columnStates }: CustomEvent<ColumnState[]>) {
+    const updatedColumnStates = columnStates.filter(columnState => columnState.colId !== 'actions');
+
+    viewUpdateActivityTables({ columnStates: updatedColumnStates }, activityTableId);
+  }
+
+  function onHideAllColumns() {
```

Comment:
I'm wondering if we should combine `onHideAllColumns` and `onShowAllColumns` to a function called `onShowOrHideAllColumns(hide: boolean)`

Reply:
I like having them explicitly separate so you don't have to necessarily use the event object, but I don't have a strong opinion one way or the other.

Reply:
but i can see that it's a lot of duplicate code...i'll just combine the two